### PR TITLE
Add Kubernetes the Hard Way Install File

### PR DESCRIPTION
**Pull request by nathanjsweet**
Signed-off-by: Nate Sweet <nathanjsweet@pm.me>

Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Thanks for contributing!

Given a Cilium-specific "Kubernetes The Hard Way" install guide, we need a file that works with that specific installation guide to make it easy to use.

### CODE REVIEW

**Comment by joestringer** [comment]
Quick question, how do we intend to keep this up-to-date?

This seems like something we should be able to generate from Helm in a similar way to the `quick-install.yaml`. That way, `make -C install/kubernetes` would update it so it wouldn't rot.

**Comment by michi-covalent** [comment]
agree with @joestringer it would be good to add a make target like {quick,experimental}_install.yaml.

also, could you also point me to the "Kubernetes The Hard Way" install guide?

File: install/kubernetes/kubernetes-the-hard-way-install.yaml [hunk 1]
```diff
@@ -0,0 +1,643 @@
...
+          readOnly: true
+      hostNetwork: true
+      restartPolicy: Always
+      priorityClassName: system-cluster-critical
+      volumes:
+        # To hold the kubeconfig file
+      - hostPath:
+          path: /etc/cilium
+          type: DirectoryOrCreate
+        name: cilium-etc
```
Comment (aanm) [inline install/kubernetes/kubernetes-the-hard-way-install.yaml:639]: Why is this required? All necessary configuration to access the cluster should be installed in the pods automatically by kubelet.

**Review by aanm — CHANGES_REQUESTED** [review]
Ideally this file should be created automatically in a similar way we are doing with the `install/kubernetes/experimental-install.yaml`

**Comment by joestringer** [comment]
Converting to draft so it doesn't show up in people's review queues. When the feedback has been addressed, please mark `Ready for review` using the button near the bottom of the page.

**Pull request closed without merging**